### PR TITLE
fix: readonly nonce input

### PR DESCRIPTION
**Pull request by schmanu**
## What it solves
Improves readability of nonce input in readonly mode + reduces the padding of the input field.
- change nonce input for signed txs to readonly.
- only add `padding-right` to input if recommendedNonce button is shown.

## Screenshots
- Edit mode with reset to recommended nonce button
Screenshot 2023-07-07 at 11 17 57

- Edit mode without reset to recommended nonce button
Screenshot 2023-07-07 at 11 21 41

- Readonly mode for queued & signed tx
Screenshot 2023-07-07 at 11 27 35

## Checklist
* [x] I've tested the branch on mobile 📱
* [x] I've documented how it affects the analytics (if at all) 📊
* [x] I've written a unit/e2e test for it (if applicable) 🧑‍💻

### CODE REVIEW

File: src/components/tx-flow/common/TxNonce/index.tsx [hunk 1]
```diff
@@ -116,7 +119,7 @@ const TxNonceForm = ({ nonce, recommendedNonce }: { nonce: number; recommendedNo
               }
             }}
             options={previousNonces}
-            disabled={readOnly}
+            readOnly={readOnly}
```
Comment (katspaugh) [inline src/components/tx-flow/common/TxNonce/index.tsx:122]: I liked your initial idea to just show a span in this case.

**Review by usame-algan — COMMENTED** [review]
I would suggest we remove the input border when its readOnly or change it so its more "static". Right now the border changes width when clicking into the input. We could also render a span instead like it was suggested

File: src/components/tx-flow/common/TxNonce/index.tsx [hunk 2]
```diff
@@ -136,21 +144,26 @@ const TxNonceForm = ({ nonce, recommendedNonce }: { nonce: number; recommendedNo
...
+                    className={classNames([
+                      css.input,
+                      {
+                        [css.withAdornment]: showRecommendedNonceButton,
+                      },
+                    ])}
                     sx={{
-                      minWidth: `clamp(1ch, ${field.value.length}ch, 200px)`,
+                      minWidth: `clamp(1ch, calc(${field.value.toString().length}ch${
+                        showRecommendedNonceButton ? ' + 28px' : ' + 4px'
```
Comment (schmanu) [inline src/components/tx-flow/common/TxNonce/index.tsx:165]: I couldn't figure out a more elegant way to do handle the dynamic widths :/

File: src/components/tx-flow/common/TxNonce/index.tsx [hunk 3]
```diff
@@ -102,6 +103,14 @@ const TxNonceForm = ({ nonce, recommendedNonce }: { nonce: number; recommendedNo
         },
       }}
       render={({ field, fieldState }) => {
+        if (readOnly) {
+          return <>{nonce}</>
+        }
+
+        console.log(field, field.value)
```
Comment (usame-algan) [inline src/components/tx-flow/common/TxNonce/index.tsx:110]:
```suggestion

```

**Review by usame-algan — APPROVED** [review]

File: src/components/tx-flow/common/TxNonce/index.tsx (hunk 2 above)
Comment (usame-algan) [reply src/components/tx-flow/common/TxNonce/index.tsx:165]: If we want to have the input as small as possible we need to take this into account so I think its fine.

File: src/components/tx-flow/common/TxNonce/index.tsx (hunk 3 above)
Comment (schmanu) [reply src/components/tx-flow/common/TxNonce/index.tsx]: whoops. Return of the classic console log leftover 🤦

**Comment by katspaugh** [comment]
It's almost as per the design at this point, the only thing lacking is bold.
Screenshot 2023-07-07 at 12 21 28

**Pull request merged**